tests/integration/security/rbac/v2/basic/basic_test.go

```diff
@@ -15,146 +15,463 @@
 package basic
 
 import (
+	"fmt"
 	"testing"
 	"time"
 
-	"istio.io/istio/pkg/test/framework/label"
-	"istio.io/istio/tests/integration/security/rbac/util"
-
+	"istio.io/istio/pilot/pkg/model"
+	"istio.io/istio/pkg/test/echo/common/scheme"
 	"istio.io/istio/pkg/test/framework"
-	"istio.io/istio/pkg/test/framework/components/apps"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
 	"istio.io/istio/pkg/test/framework/components/environment"
-	"istio.io/istio/pkg/test/framework/components/environment/kube"
-	"istio.io/istio/pkg/test/framework/components/galley"
-	"istio.io/istio/pkg/test/framework/components/istio"
-	"istio.io/istio/pkg/test/framework/components/pilot"
-	"istio.io/istio/pkg/test/util/connection"
-	"istio.io/istio/pkg/test/util/policy"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/util/file"
 	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/tmpl"
+	"istio.io/istio/tests/integration/security/rbac/util"
+	"istio.io/istio/tests/integration/security/util/connection"
 )
 
 const (
 	rbacClusterConfigTmpl = "testdata/istio-clusterrbacconfig.yaml.tmpl"
 	rbacV2RulesTmpl       = "testdata/istio-rbac-v2-rules.yaml.tmpl"
 )
 
-var (
-	inst          istio.Instance
-	isMtlsEnabled bool
-)
-
-func TestMain(m *testing.M) {
-	framework.
-		NewSuite("rbac_v2", m).
-		// TODO(pitlv2109: Turn on the presubmit label once the test is stable.
-		RequireEnvironment(environment.Kube).
-		Label(label.CustomSetup).
-		SetupOnEnv(environment.Kube, istio.Setup(&inst, setupConfig)).
-		Run()
-}
-
-func setupConfig(cfg *istio.Config) {
-	if cfg == nil {
-		return
-	}
-	isMtlsEnabled = cfg.IsMtlsEnabled()
-	cfg.Values["sidecarInjectorWebhook.rewriteAppHTTPProbe"] = "true"
-}
-
 func TestRBACV2(t *testing.T) {
-	ctx := framework.NewContext(t)
-	defer ctx.Done(t)
-	ctx.RequireOrSkip(t, environment.Kube)
-
-	env := ctx.Environment().(*kube.Environment)
-	g := galley.NewOrFail(t, ctx, galley.Config{})
-	p := pilot.NewOrFail(t, ctx, pilot.Config{
-		Galley: g,
-	})
-	appInst := apps.NewOrFail(t, ctx, apps.Config{Pilot: p, Galley: g})
-
-	appA, _ := appInst.GetAppOrFail("a", t).(apps.KubeApp)
-	appB, _ := appInst.GetAppOrFail("b", t).(apps.KubeApp)
-	appC, _ := appInst.GetAppOrFail("c", t).(apps.KubeApp)
-	appD, _ := appInst.GetAppOrFail("d", t).(apps.KubeApp)
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			ns := namespace.NewOrFail(t, ctx, "rbacv1", true)
+			ports := []echo.Port{
+				{
+					Name:        "http",
+					Protocol:    model.ProtocolHTTP,
+					ServicePort: 80,
+				},
+				{
+					Name:        "tcp",
+					Protocol:    model.ProtocolTCP,
+					ServicePort: 90,
+				},
+			}
+			a := echoboot.NewOrFail(t, ctx, echo.Config{
```

Comment:
Sorry for the late comment, I was out yesterday. Why can't we get rid of this boilerplate here can use function with parameters instead? It seems like a lot of similar code in many tests. Also, why doesn't `a` have ServiceAccount?